CDCgov/NEDSS-Modernization
# CNFT1-1517 Migrate local dev to 6.0.15

## Description

To migrate the local development database to 6.0.15, the migration scripts found [here](https://github.com/cdcent/NEDSSDB/tree/master/NBS_DB/db/Disk_Database/WINDOWS/DB%20Setup%20Scripts-Release%20Upgrade/MS%20SQL) were applied sequentially.

Following the application of these scripts, the database was "pared" down, removing large quantities of unused data. Due to the removal of this data we are now able to persist the `.bak` files within Github.

The `test-db` has been removed. The `nbs-mssql` image will be used for testing. The github action to build and store the test-db has been changed to point to the `nbs-mssql` Dockerfile. The Dockerfile has also been modified to create a restored database image instead of having to run the restore scripts on creation of any new container.

Updates the `sonar.yaml` action to zip the test reports prior to uploading. The `actions/upload-artifact@v2` does not zip and sends a different http request for each file. Zipping the test report and uploading the zipped file reduces the upload time from ~1 minute to < 1 second.

The build process has been updated to utilize a `builder` in the wildfly container. 

A new set of scripts `build_all.sh`, `build_classic.sh`, and `build_modernized.sh` were created to simplify the process of standing up a local environment.

With the migration to 6.0.15, `ant` builds are no longer supported

## Issues

[CNFT1-1517](https://cdc-nbs.atlassian.net/browse/CNFT1-1517)

## Steps to verify

1. Delete local containers
2. Run the provided `build_all.sh` script
3. Verify database is restored and `NBS_ODSE.NBS_configuration` table displays version `6.0.15`
4. Verify application is accessible using `msa` or `superuser`



[CNFT1-1517]: https://cdc-nbs.atlassian.net/browse/CNFT1-1517?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
